diff --git a/cores/piduino/Arduino.h b/cores/piduino/Arduino.h
index 9ab4451..6f6f6fe 100644
--- a/cores/piduino/Arduino.h
+++ b/cores/piduino/Arduino.h
@@ -25,6 +25,7 @@
 #include "bcm2835_registers.h"
 #include <math.h>
 #include <pthread.h>
+#include <stdbool.h>
 #include "pgmspace.h"
 
 #ifdef __cplusplus
@@ -109,7 +110,7 @@ void yield(void);
 #define millis() (unsigned long)(STCV / 1000)
 
 typedef unsigned int word;
-typedef uint8_t boolean;
+typedef bool boolean;
 typedef uint8_t byte;
 
 //under millisecond delayMicroseconds halts the CPU
